core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -132,6 +138,60 @@ public ClientFactoryBuilder workerGroup(EventLoopGroup workerGroup, boolean shut
         return this;
     }
 
+    /**
+     * Sets the factory that creates an {@link EventLoopScheduler} which is responsible for assigning an
+     * {@link EventLoop} to handle a connection to the specified {@link Endpoint}.
+     */
+    public ClientFactoryBuilder eventLoopSchedulerFactory(
+            Function<? super EventLoopGroup, ? extends EventLoopScheduler> eventLoopSchedulerFactory) {
+        checkState(maxNumEventLoopsPerEndpoint == 0 && maxNumEventLoopsMap.isEmpty(),
+                   "Cannot set eventLoopSchedulerFactory when maxEventLoop per endpoint is specified.");
+        this.eventLoopSchedulerFactory = requireNonNull(eventLoopSchedulerFactory, "eventLoopSchedulerFactory");
+        return this;
+    }
+
+    /**
+     * Sets the maximum number of {@link EventLoop}s which will be used to handle connections to all
+     * {@link Endpoint}s except the ones specified by {@link #maxNumEventLoopsPerEndpoint(Endpoint, int)} or
+     * {@link #maxNumEventLoopsPerEndpoint(String, int)}.
+     */
+    public ClientFactoryBuilder maxNumEventLoopsPerEndpoint(int maxNumEventLoopsPerEndpoint) {
+        checkArgument(maxNumEventLoopsPerEndpoint > 0,
+                      "maxNumEventLoopsPerEndpoint: %s (expected: > 0)", maxNumEventLoopsPerEndpoint);
+        checkState(eventLoopSchedulerFactory == null,
+                   "maxNumEventLoopsPerEndpoint() and eventLoopSchedulerFactory() are mutually exclusive.");
+        this.maxNumEventLoopsPerEndpoint = maxNumEventLoopsPerEndpoint;
+        return this;
+    }
+
+    /**
+     * Sets the maximum number of {@link EventLoop}s which will be used to handle connections to the specified
+     * {@link Endpoint}.
+     * This method is a shortcut of:
+     * <pre>{@code
+     * String authority = endpoint.authority();
+     * maxNumEventLoopsPerEndpoint(authority, maxNumEventLoops);
+     * }</pre>
+     */
+    public ClientFactoryBuilder maxNumEventLoopsPerEndpoint(Endpoint endpoint, int maxNumEventLoops) {
+        requireNonNull(endpoint, "endpoint");
+        return maxNumEventLoopsPerEndpoint(endpoint.authority(), maxNumEventLoops);
+    }
+
+    /**
+     * Sets the maximum number of {@link EventLoop}s which will be used to handle connections to the specified
+     * {@code authority}.
+     */
+    public ClientFactoryBuilder maxNumEventLoopsPerEndpoint(String authority, int maxNumEventLoops) {
+        requireNonNull(authority, "authority");
+        checkState(eventLoopSchedulerFactory == null,
+                   "maxNumEventLoopsPerEndpoint() and eventLoopSchedulerFactory() are mutually exclusive.");
+        checkArgument(maxNumEventLoops > 0,
+                      "maxNumEventLoopsPerEndpoint for '%s': %s (expected: > 0)", authority, maxNumEventLoops);
+        maxNumEventLoopsMap.put(authority, maxNumEventLoops);
```

Comment:
- How about checking if `authority` is a valid `Endpoint` expression using `Endpoint.parse()`?
- We could also make `maxNumEventLoopsPerEndpoint(String, int)` a shortcut to `maxNumEventLoopsPerEndpoint(Endpoint.parse(authority), ...)` to avoid double parsing.